cmd/titus-imds-cni/env.go

```diff
@@ -109,9 +107,36 @@ func extractEnv(prev *current.Result, pod *v1.Pod) (map[string]string, error) {
 	env["TITUS_IAM_ROLE"] = pod.Annotations[mt.IamRoleArnAnnotation]
 	env["X_FORWARDED_FOR_BLOCKING_MODE"] = pod.Annotations[mt.XForwardedForBlockingModeAnnotation]
 
+	c := getUserContainer(pod)
+	if mEnvVal, ok := getEnvVal(c, mt.TitusMetatronVariableName); ok {
+		env[mt.TitusMetatronVariableName] = mEnvVal
+	}
+
 	return env, nil
 }
 
+func getEnvVal(c *v1.Container, name string) (string, bool) {
+	for _, e := range c.Env {
+		if e.Name == name {
+			return e.Value, true
+		}
+	}
+
+	return "", false
+}
+
+func getUserContainer(pod *v1.Pod) *v1.Container {
+	firstContainer := pod.Spec.Containers[0]
+	for _, c := range pod.Spec.Containers {
+		if c.Name == pod.Name {
+			ctr := c
+			return &ctr
+		}
+	}
+
+	return &firstContainer
```

Comment:
Why the fallback assumption that the first container is the user container if we don't find a podName == containerName match?  Is it better to fail?

Reply:
Picking the first container has been the behaviour so far in all of our code, so I'm preserving that for backwards compatibility.